kmath-core/src/commonMain/kotlin/space/kscience/kmath/linear/MatrixBuilder.kt

```diff
@@ -1,46 +1,43 @@
 package space.kscience.kmath.linear
 
-import space.kscience.kmath.nd.Structure2D
-import space.kscience.kmath.structures.Buffer
-import space.kscience.kmath.structures.BufferFactory
-import space.kscience.kmath.structures.asBuffer
-
-public class MatrixBuilder(public val rows: Int, public val columns: Int) {
-    public operator fun <T : Any> invoke(vararg elements: T): Matrix<T> {
+import space.kscience.kmath.misc.UnstableKMathAPI
+import space.kscience.kmath.operations.Ring
+
+public class MatrixBuilder<T : Any, A : Ring<T>>(
```

Comment:
A very strange class. Were there any other options?

Reply:
They could be added later like `matrix(2,2).diagonal(2.0)`. This class allows creating a variety of builders abstracted away from LinearSpace specifics.